pkg/storage/unified/sql/queries_test.go

```diff
@@ -1,353 +1,184 @@
+								Group:     "gg",
+								Resource:  "rr",
+								Name:      "name",
+							},
+						},
 					},
 				},
 			},
-		},
-
-		sqlResourceInsert: {
-			{
-				Name: "insert into resource",
-				Data: &sqlResourceRequest{
-					SQLTemplate: new(sqltemplate.SQLTemplate),
-					WriteEvent: resource.WriteEvent{
-						Key: &resource.ResourceKey{},
-					},
-				},
-				Expected: expected{
-					"resource_insert_mysql_sqlite.sql": dialects{
-						sqltemplate.MySQL,
-						sqltemplate.SQLite,
+			sqlResourceInsert: {
+				{
+					Name: "simple",
+					Data: &sqlResourceRequest{
+						SQLTemplate: new(sqltemplate.SQLTemplate),
+						WriteEvent: resource.WriteEvent{
+							Key: &resource.ResourceKey{
+								Namespace: "nn",
+								Group:     "gg",
+								Resource:  "rr",
+								Name:      "name",
+							},
+							Type:       resource.WatchEvent_ADDED,
+							PreviousRV: 123,
+						},
 					},
 				},
 			},
-		},
-
-		sqlResourceUpdate: {
-			{
-				Name: "single path",
-				Data: &sqlResourceRequest{
-					SQLTemplate: new(sqltemplate.SQLTemplate),
-					WriteEvent: resource.WriteEvent{
-						Key: &resource.ResourceKey{},
-					},
-				},
-				Expected: expected{
-					"resource_update_mysql_sqlite.sql": dialects{
-						sqltemplate.MySQL,
-						sqltemplate.SQLite,
+			sqlResourceUpdate: {
+				{
+					Name: "single path",
+					Data: &sqlResourceRequest{
+						SQLTemplate: new(sqltemplate.SQLTemplate),
+						WriteEvent: resource.WriteEvent{
+							Key: &resource.ResourceKey{},
+						},
 					},
 				},
 			},
-		},
-
-		sqlResourceRead: {
-			{
-				Name: "without resource version",
-				Data: &sqlResourceReadRequest{
-					SQLTemplate: new(sqltemplate.SQLTemplate),
-					Request: &resource.ReadRequest{
-						Key: &resource.ResourceKey{},
-					},
-					readResponse: new(readResponse),
-				},
-				Expected: expected{
-					"resource_read_mysql_sqlite.sql": dialects{
-						sqltemplate.MySQL,
-						sqltemplate.SQLite,
+			sqlResourceRead: {
+				{
+					Name: "without_resource_version",
+					Data: &sqlResourceReadRequest{
+						SQLTemplate: new(sqltemplate.SQLTemplate),
+						Request: &resource.ReadRequest{
+							Key: &resource.ResourceKey{},
+						},
+						readResponse: new(readResponse),
 					},
 				},
 			},
-		},
 
-		sqlResourceList: {
-			{
-				Name: "filter on namespace",
-				Data: &sqlResourceListRequest{
-					SQLTemplate: new(sqltemplate.SQLTemplate),
-					Request: &resource.ListRequest{
-						Limit: 10,
-						Options: &resource.ListOptions{
-							Key: &resource.ResourceKey{
-								Namespace: "ns",
+			sqlResourceList: {
+				{
+					Name: "filter_on_namespace",
+					Data: &sqlResourceListRequest{
+						SQLTemplate: new(sqltemplate.SQLTemplate),
+						Request: &resource.ListRequest{
+							Limit: 10,
+							Options: &resource.ListOptions{
+								Key: &resource.ResourceKey{
+									Namespace: "ns",
+								},
 							},
 						},
 					},
 				},
-				Expected: expected{
```

Comment:
The big difference between this generic test and the one here is that you do not specifiy explicit  dialects and the filename.  In the new one it does (by default) MySQL, SQLite, and postgres and finds a template for each one